skimage/filters/_blur_effect.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+import scipy.ndimage as ndi
+
+from ..color import rgb2gray
+from . import sobel
+from ..util import img_as_float
+
+
+__all__ = ['blur_effect']
+
+
+def blur_effect(image, h_size=11, channel_axis=None):
+    """
+    Compute a metric that indicates the strength of blur in an image (0 for no
+    blur, 1 for maximal blur).
+
+    Parameters
+    ----------
+    image : ndarray
+        RGB or grayscale nD image. The input image is converted to grayscale
+        before computing the blur metric.
+    h_size : int, optional
+        Size of the re-blurring filter. Default is 11.
+    channel_axis : int or None, optional
+        If None, the image is assumed to be grayscale (single-channel).
+        Otherwise, this parameter indicates which axis of the array
+        corresponds to color channels.
+
+    Returns
+    -------
+    blur : float (0 to 1)
+        Blur metric: the maximum of blur metrics along all axes.
+    blur_table : list of floats
+        The i-th element is the blur metric along the i-th axis.
+
+    Notes
+    -----
+    `h_size` must keep the same value in order to compare results between
+    images. Most of the time, the default size (11) is enough. This means that
+    the metric can clearly discriminate blur up to an average 11x11 filter; if
+    blur is higher, the metric still gives good results but its values tend
+    towards an asymptote.
+
+    References
+    ----------
+    .. [1] Frederique Crete, Thierry Dolmiere, Patricia Ladret, and Marina
+       Nicolas "The blur effect: perception and estimation with a new
+       no-reference perceptual blur metric" Proc. SPIE 6492, Human Vision and
+       Electronic Imaging XII, 64920I (2007)
+       https://hal.archives-ouvertes.fr/hal-00232709
+       :DOI:`10.1117/12.702790`
+    """
+    # number of non-channel axes
+    n_axes = image.ndim
+
+    if channel_axis is not None:
+        try:
+            # ensure color channels are in the final dimension
+            image = np.moveaxis(image, channel_axis, -1)
+        except np.AxisError:
+            print('channel_axis must be one of the image array dimensions')
+            raise
+        except TypeError:
+            print('channel_axis must be an integer')
+            raise
+        image = rgb2gray(image)
+        n_axes = n_axes - 1
+    image = img_as_float(image)
+    shape = image.shape
+    B = []
+
+    for a in range(n_axes):
+        filt_im = ndi.uniform_filter1d(image, h_size, axis=a)
+        im_sharp = np.abs(sobel(image, axis=a))
+        im_blur = np.abs(sobel(filt_im, axis=a))
+        T = np.maximum(0, im_sharp - im_blur)
+        slices = [slice(2, shape[ax] - 1) for ax in range(n_axes)]
+        M1 = np.sum(im_sharp[tuple(slices)])
+        M2 = np.sum(T[tuple(slices)])
+        B.append(np.abs((M1 - M2)) / M1)
+
+    return np.max(B), B
```

Comment:
What about introducing a `metric` parameter which is a reduction function (with default to `np.max`), to be applied to `B`?
This allows to use what ever reduction to `B` (`mean`, `std`...), to return plain `B` when it is set to `None` and avoids returning `B` along with its maximum as it is made here?
```suggestion
    return B if metric is None else metric(B)
```

Reply:
Plain `B` is available as the second returned value, so the user can always access it and aggregate it:
```py
_, plain_B = blur_effect(my_image)
my_blur_metric = np.mean(plain_B)
```
for example. The 'default' to `np.max` is based on the referenced paper: https://hal.archives-ouvertes.fr/hal-00232709/document where the metric is validated against human perception.

Reply:
Yes, I see that plain ``B` is returned, and that's what I find not elegant! Most users will use only one of the two returned values...
So I though about something similar to the `reduce_func` argument in [skimage.measure.profile_line](https://scikit-image.org/docs/dev/api/skimage.measure.html#skimage.measure.profile_line)....

Reply:
Ok, I see! I admit that returning multiple values isn't the most elegant... Let me add an optional `reduce_func` then!

Reply:
Excellent, thank you @mkcor :wink:

Reply:
Here it is: 62dd792f0870b2f2d8cf33b1d61bf2a18e5515a8 :slightly_smiling_face: 
Thank you for the thorough review, @rfezzani, it's been really helpful! :muscle: